Statement.php

```diff
@@ -147,17 +156,21 @@ public function errorInfo() {
 
     /**
      * {@inheritdoc}
+     *
+     * @SuppressWarnings("PHPMD.StaticAccess")
      */
     public function execute($params = null) {
-        $rawRequest = $this->mysqlToRequest->transform($this->query, $this->params);
-        $request    = $this->authStrategy->transformRequest($rawRequest);
-        $restClient = $this->restClientFactory->createOne($request->getCurlOptions());
+        $transformedQuery = SqlQuery::quoteUrl(SqlQuery::setParams($this->query, $params !== null ? $params : $this->params));
+        $rawRequest       = $this->mysqlToRequest->transform($transformedQuery);
+        $request          = $this->authStrategy->transformRequest($rawRequest);
+
+        $response = $this->restClient->send($request);
 
-        $method     = strtolower($request->getMethod());
-        $response   = $method === HttpMethods::GET || $method === HttpMethods::DELETE ? $restClient->$method($request->getUrlAndQuery()) : $restClient->$method($request->getUrlAndQuery(), $request->getPayload());
-        $statusCode = $response->getStatusCode();
+        $this->result = Result::create($transformedQuery, $this->formatter->decode($response->getContent()));
+        $this->id     = !empty($this->result['id']) ? $this->result['id'] : null;
+        krsort($this->result);
 
-        return $statusCode === 200 || ($method === HttpMethods::DELETE && $statusCode === 204) ? $this->onSuccess($response, $method) : $this->onError($request, $response);
```

Comment:
So, also die sind richtig klasse. Ich habe das jetzt mit try catch probiert und dann this.errorCode = e.code usw... und halt return true, return false. Ergebnis: Das Return-Value wird von Doctrine überhaupt nicht beachtet. Solange da keine Exception geworfen wird läuft alles seinen Gang.

Danach habe ich geguckt wie der MySQL (PDO) Treiber das implementiert hat (https://github.com/doctrine/dbal/blob/master/lib/Doctrine/DBAL/Driver/PDOStatement.php). Lustigerweise returned der bool, aber wirft im Fehlerfall eine PDOException.

Mein Vorschlag wäre daher, das in etwa so zu lösen wie es eh schon war:

```php
    /**
     * {@inheritdoc}
     * @throws RequestFailedException
     *
     * @SuppressWarnings("PHPMD.StaticAccess")
     */
    public function execute($params = null) {
        $transformedQuery = SqlQuery::quoteUrl(SqlQuery::setParams($this->query, $params !== null ? $params : $this->params));
        $rawRequest       = $this->mysqlToRequest->transform($transformedQuery);
        $request          = $this->authStrategy->transformRequest($rawRequest);

        $response = $this->restClient->send($request);
        $result   = new Result($transformedQuery, $this->formatter->decode($response->getContent()));

        $this->result = $result->get();
        $this->id     = $result->id();

        return true;
    }
```